src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -242,12 +247,42 @@ private ApiEvent InitializeApiEvent(string accountId)
             apiEvent.TokenType = AuthenticationRequestParameters.AuthenticationScheme.TelemetryTokenType;
             apiEvent.AssertionType = GetAssertionType();
 
+            UpdateCallerSdkDetails(apiEvent);
+
             // Give derived classes the ability to add or modify fields in the telemetry as needed.
             EnrichTelemetryApiEvent(apiEvent);
 
             return apiEvent;
         }
 
+        private void UpdateCallerSdkDetails(ApiEvent apiEvent)
+        {
+            string callerSdkId;
```

Comment:
is this an id or the SDK name?

Reply:
No Api id of the api used by the caller SDK